### PR TITLE
feature/implementation of view posts page card

### PR DESCRIPTION
**Pull request by RICHIEm8**

### CODE REVIEW

File: src/client/App.tsx [hunk 1]
```diff
@@ -1,10 +1,34 @@
...
-const App: React.FC = () => {
-  return <div>reddit clone</div>;
-};
+const { Header, Footer, Sider, Content } = Layout;
 
-export default App;
+export const App: React.FC = () => {
+  const [data, loading] = useFetch("http://localhost:3000/api/posts");
+  if (loading) {
+    return <h1>Loading</h1>;
```
Comment (dbousamra) [inline src/client/App.tsx]: Hmm. I think you probably want to return Loading inside the rest of your layout. 

Does that make sense?

File: src/client/App.tsx [hunk 2]
```diff
@@ -1,10 +1,34 @@
...
+    return (
+      <div>
+        <Layout>
+          <Header>Reddit Clone</Header>
+          <Layout>
+            <Content>
+              Content
+              <PostsList posts={data} />
+            </Content>
+            <Sider>Sider</Sider>
```
Comment (dbousamra) [inline src/client/App.tsx]: Are you using the Sider and Footer?

File: src/client/components/PostDetail.tsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+export interface PostDetailProps {
+  post: Post;
+}
+
+export const PostDetail: React.SFC<PostDetailProps> = (
+  props: PostDetailProps
+) => {
+  const { post } = props;
+  return (
+    <div>
```
Comment (dbousamra) [inline src/client/components/PostDetail.tsx]: Do you want to use AntD components here too?

File: src/client/index.html [hunk 4]
```diff
@@ -4,7 +4,7 @@
     <meta charset="UTF-8" />
     <meta name="viewport" content="width=device-width, initial-scale=1.0" />
     <meta http-equiv="X-UA-Compatible" content="ie=edge" />
-    <link rel="stylesheet" href="index.less" />
+    <!-- <link rel="stylesheet" href="index.less" /> -->
```
Comment (dbousamra) [inline src/client/index.html]: Why is this here?

File: src/client/index.less [hunk 5]
```diff
@@ -1,8 +1,21 @@
-body {
-  font-family: sans-serif;
-  background-color: gray;
-}
+@import "~antd/dist/antd.css";
```
Comment (dbousamra) [inline src/client/index.less]: Why is this here if you are importing the css in the App.tsx?

File: src/client/index.less [hunk 6]
```diff
@@ -1,8 +1,21 @@
...
 
-h1 {
-  color: white;
+.ant-layout-header,
+.ant-layout-footer {
+  background: #7dbcea;
+  color: #fff;
+}
+.ant-layout-footer {
```
Comment (dbousamra) [inline src/client/index.less]: What are these classes for?

File: src/client/App.tsx (hunk 2 above)
Comment (RICHIEm8) [reply src/client/App.tsx]: not yet but we will be. having trouble actually implementing that AntD layout to be honest

File: src/client/components/PostDetail.tsx (hunk 3 above)
Comment (RICHIEm8) [reply src/client/components/PostDetail.tsx]: maybe, havent actually properly implemented AntD yet

File: src/client/index.html (hunk 4 above)
Comment (RICHIEm8) [reply src/client/index.html]: was there to begin with, was just fiddling with it trying to work the css but I think we can just delete it since we are already importing the css inside our app.tsx

File: src/client/index.less (hunk 5 above)
Comment (RICHIEm8) [reply src/client/index.less]: we are just following the AntD docs on how to import for the layout, etc

File: src/client/index.less (hunk 6 above)
Comment (RICHIEm8) [reply src/client/index.less]: its what gives the layout the colours and css

File: src/client/App.tsx (hunk 1 above)
Comment (sbousamra) [reply src/client/App.tsx]: I agree. You should always have a Layout i think, and what's rendered inside the Layout depends on if loading or not
